### PR TITLE
[DX-1501] Large update to event handling

### PR DESCRIPTION
**Pull request by andyo-tyk**
https://tyktech.atlassian.net/browse/DX-1501

**This doesn't need DX review until peer review is complete**

### Preview Link
https://deploy-preview-5051--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/report-monitor-trigger-events/

### Description
Restructured event handling, moved some other pages for clarity (e.g. logging -> observability)
Added new page for existing logging event handler (previously undocumented)
Overhauled webhook event handler docs including new support in Tyk OAS

### CODE REVIEW

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
-date: 2017-03-24T12:36:32Z
-title: Webhooks
-tags: ["Webhooks", "Events"]
-description: "Using the Tyk webhook event handler"
-menu:
-  main:
-    parent: "Report, Monitor and Trigger Events"
-weight: 3 
+date: 20224-07-10
+title: Event handling with webhooks
```
Comment (lghiur) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:3]: in the UI this is too long IMO. Just webhooks would do it

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
 ---
 
-## Set up an Event Webhook with an API Definition
+A webhook is a mechanism for real-time, event-driven communication between different systems or applications over the internet. It is an HTTP callback, typically an HTTP POST request that occurs when something happens. Webhooks are real-time, automated and lightweight. Notifications are sent immediately when events occur without the need for the receiving service to poll.
 
-In order to add extensibility and interoperability to Tyk, a new webhook event handler has been added. This allows a fixed payload (which can be customised) to be delivered to any open endpoint.
+In the context of Tyk Gateway, webhooks are event handlers that can be registered against API Events. The webhook will be triggered when the corresponding event is fired and will send a customisable fixed payload to any open endpoint.
 
-The webhook handler allows the configuration of the method, body, header values, and target URL and makes use of Go templates to expose the event metadata.
+## When to use webhook event handlers
```
Comment (lghiur) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:14]: can we make it clear that these are just a few exapmples?

**Review by lghiur — APPROVED** [review]

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md (hunk 1 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:3]: The key terminology that I've adopted throughout this PR is that it refers to event handling... and in this case, with webhooks. I want to avoid any risk of confusion with API webhooks - here we are talking only about system events, not business logic.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md (hunk 2 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:14]: I've adopted this approach in various pages - I guess we could add a short statement under this header to say "there are many occasions when you might use webhooks for event handling, here are a few examples...". Something like this?

Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:14]: I've added this line.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
 
-Tyk JSVM event handlers have access to the Tyk JSVM API, which gives access to the session object store, and enables the ability for making HTTP calls outside of the Tyk environment, this is particularly useful if you want to interface with another API with a complex request/response cycle.
+Custom event handlers are currently only supported by Tyk Classic APIs.
+{{< /note >}}
 
-### Creating an Event Handler
+### Creating a custom event handler
 
-Creating an event handler is very similar to creating middleware, simply invoke the correct constructors with a closure in the TykJS namespace:
+A custom event handler consists of a function that accepts two variables ()`event` and `context`) and has no return value.
```
Comment (jeffy-mathew) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md]:
```suggestion
A custom event handler consists of a function that accepts two variables (`event` and `context`) and has no return value.
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md [hunk 4]
```diff
@@ -1,29 +1,46 @@
...
-*   `BreakerTriggered`: This event will be created when either a `BreakerTripped`, or a `BreakerReset` event occurs; a status code in the metadata passed to the webhook will indicate which of these events was triggered.
-*   `BreakerTripped`: When a circuit breaker on a path trips and a service is taken offline.
-*   `BreakerReset`: When the circuit breaker comes back on-stream.
-*   `TokenCreated`: Executed when a token is created
-*   `TokenUpdated`: Executed when a token is changed
-*   `TokenDeleted`: Executed when a token is deleted
+- `RatelimitExceeded`: the rate limit has been exceeded for a specific key
+- `OrgRateLimitExceeded`: the rate limit has been exceeded for a specific organisation
+- `RateLimitSmoothingUp`: the [intermediate rate limit allowance]({{< ref "getting-started/key-concepts/rate-limiting#rate-limit-smoothing" >}}) has been increased for a specific key
+- `RateLimitSmoouthingDown`: the [intermediate rate limit allowance]({{< ref "getting-started/key-concepts/rate-limiting#rate-limit-smoothing" >}}) has been decreased for a specific key
```
Comment (jeffy-mathew) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md]:
```suggestion
- `RateLimitSmoothingDown`: the [intermediate rate limit allowance]({{< ref "getting-started/key-concepts/rate-limiting#rate-limit-smoothing" >}}) has been decreased for a specific key
```

**Review by jeffy-mathew — APPROVED** [review]

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md [hunk 5]
```diff
@@ -1,29 +1,46 @@
...
+- `BreakerTripped`: a circuit breaker on a path has tripped and been taken offline
+- `BreakerReset`: a circuit breaker has reset and the path is available again
+- `BreakerTriggered`: a circuit breaker has changed state, this is generated when either a `BreakerTripped`, or a `BreakerReset` event occurs; a status code in the metadata passed to the webhook will indicate which of these events was triggered
+
+### Uptime events
+
+- `HostDown`: the uptime checker has found that a host is down/not available
+- `HostUp`: the uptime checker has found that a host is available again after being offline
+
+## Token lifecycle events
```
Comment (andyo-tyk) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md]:
```suggestion
### Token lifecycle events
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md [hunk 6]
```diff
@@ -1,18 +1,50 @@
...
+
+For this, Tyk provides a dedicated [quota monitoring]({{< ref "basic-config-and-security/report-monitor-trigger-events/monitors" >}}) capability that can be configured to trigger a dedicated monitor event handler when the API usage exceeds different thresholds approaching the quota limit.
+
+### Custom events
 
 The event subsystem has been designed to be easily extensible, so the community can provide additional event handlers (and add events) to the Tyk codebase or they can be compiled into the version branch very easily for custom builds.
+
+## Handling events with Tyk
+
+In response to an API, token lifecycle or custom event occuring, at the API-level you can:
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md]: What  does it refer to when mentioning in response an API, is it an API request?

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md [hunk 7]
```diff
@@ -1,18 +1,50 @@
...
+- register a [webhook]({{< ref "basic-config-and-security/report-monitor-trigger-events/webhooks" >}}) that will call out to an external endpoint
+- send an [event log]({{< ref "product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/log-handlers" >}}) to the configured [log output]({{< ref "log-data" >}})
+- create your own [custom event handler]({{< ref "basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript" >}}) that will run in a JavaScript virtual machine on the Tyk server
+- note that **quota usage monitoring** has a [dedicated monitor]({{< ref "basic-config-and-security/report-monitor-trigger-events/monitors" >}}) to handle these events
+
+### Event metadata
+
+When an API event is fired, if there is an event handler registered for that combination of API and event then the handler will be provided with a rich set of [metadata]({{< ref "basic-config-and-security/report-monitor-trigger-events/event-data" >}}) that can be used by the external system (webhook) or custom (JavaScript) code to determine the action to be taken.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md]:
```suggestion
When an API event is fired, if there is an event handler registered for an event from that API then the handler will be provided with a rich set of [metadata]({{< ref "basic-config-and-security/report-monitor-trigger-events/event-data" >}}) that can be used by the external system (webhook) or custom (JavaScript) code to determine the action to be taken.
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md [hunk 8]
```diff
@@ -1,30 +1,31 @@
...
+title: Custom API event handlers
+tags: ["API events", "Custom handlers", "JavaScript", "event handling"]
+description: "How to create your own custom event handlers in JavaScript"
 ---
 
-## Events: JavaScript Functions
+Tyk supports you to script your own custom code in JavaScript (JS) that will be invoked in response to API events. This is executed asynchronously so you don't need to worry about it blocking the Gateway handling requests. Event handlers like this can be very powerful for automating session, user and API-level functions.
 
-### Overview
+It is important to note that unlike custom JavaScript [plugins]({{< ref "plugins/supported-languages/javascript-middleware" >}}), custom event handlers execute in a *global* JavaScript environment. This means that if you need to be careful when naming the event handlers: if you use the same event handler name for different event handling code across two APIs, only one of them will execute, as the other will be overridden when loaded.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md]:
```suggestion
It is important to note that unlike custom JavaScript [plugins]({{< ref "plugins/supported-languages/javascript-middleware" >}}), custom event handlers execute in a *global* JavaScript environment. This means that you need to be careful when naming the event handlers: if you use the same event handler name for different event handling code across two APIs, only one of them will execute, as the other will be overridden when loaded.
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md [hunk 9]
```diff
@@ -47,56 +49,59 @@ A handler consists of a function that accepts two variables called `event` and `
...
-```
+```js
 type JSVMContextGlobal struct {
   APIID string
   OrgID string
 }
 ```
 
-This can be used as follows:
+It is populated with the the API ID and Org ID of the request that your custom function can use together with the `event` metadata to interact with the Tyk REST API functions, for example:
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript.md]: Fix typo:

```suggestion
It is populated with the API ID and Org ID of the request that your custom function can use together with the `event` metadata to interact with the Tyk REST API functions, for example:
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-data.md [hunk 10]
```diff
@@ -1,33 +1,28 @@
...
+The metadata are exposed so that they can be used by the event handler (webhook or custom) using Go templating. For details of how each type of event handler can access these data, please see the appropriate section for [webhook]({{< ref "basic-config-and-security/report-monitor-trigger-events/webhooks#webhook-payload" >}}) or [custom]({{< ref "basic-config-and-security/report-monitor-trigger-events/custom-handlers-javascript#the-event-object" >}}) event handlers.
 
 ### Raw Request Data
 
-Tyk will supply a Base64 encoded representation of the original request to the event handler, if you are running a service bus or queue that stores failed, throttled or other types of requests, you can decode this object and parse it in order to re-create the original intent of the request (e.g. for post-processing).
\ No newline at end of file
+The `OriginatingRequest` metadata is a Base64-encoded wire-protocol representation of the original request to the event handler, if you are running a service bus or queue that stores failed, throttled or other types of requests, you can decode this object and parse it in order to re-create the original intent of the request (e.g. for post-processing).
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-data.md]:
```suggestion
The `OriginatingRequest` metadata is a Base64-encoded wire-protocol representation of the original request to the event handler. If you are running a service bus or queue that stores failed, throttled or other types of requests, you can decode this object and parse it in order to re-create the original intent of the request (e.g. for post-processing).
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md [hunk 11]
```diff
@@ -1,29 +1,46 @@
...
+
+### API version events
+
+- `VersionFailure`: a key has attempted access to a version of an API that it does not have permission to access
+
+### Circuit breaker events
+
+- `BreakerTripped`: a circuit breaker on a path has tripped and been taken offline
+- `BreakerReset`: a circuit breaker has reset and the path is available again
+- `BreakerTriggered`: a circuit breaker has changed state, this is generated when either a `BreakerTripped`, or a `BreakerReset` event occurs; a status code in the metadata passed to the webhook will indicate which of these events was triggered
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md:35]: Is there duplicate events created? Could _BreakerTripped_ and _BreakerReset_ events be deprecated and _BreakerTriggered_ event updated to include the details in the event payload? Or, deprecated the BreakerTriggered event?

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md [hunk 12]
```diff
@@ -1,52 +1,49 @@
...
+StatsD is a network daemon that listens for statistics, like counters and timers, sent over UDP or TCP and sends aggregates to one or more pluggable backend services. It's a simple yet powerful tool for collecting and aggregating application metrics.
 
-The Pump also requires setting the following environment variable: `TYK_INSTRUMENTATION=1`
+#### Configuring StatsD instrumentation
 
-## StatsD Keys
+To enable instrumentation for StatsD, you must set the environment variable: `TYK_INSTRUMENTATION=1` and then configure the `statsd_connection_string` field for each component.
 
-There are plenty of StatsD keys available when you enable statsd, but these are the basics:
+`statsd_connection_string` is a formatted string that specifies how to connect to the StatsD server. It typically includes information such as the host address, port number, and sometimes additional configuration options.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md:24]: Add example of statsd connection string or link to an example

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md [hunk 13]
```diff
@@ -1,52 +1,49 @@
...
+- Tyk Gateway API: `counters.<prefix>.SystemAPICall.called.count` (calls count) and `timers.<prefix>.SystemAPICall.success` (response time)
+- Tyk Dashboard API: `counters.<prefix>.SystemAPICall.SystemCallComplete.count` (requests count), `counters.<prefix>.DashSystemAPIError.*` (api error reporting)
+- Tyk Pump records: `counters.<prefix>.record.count` (number of records processed by pump)
 
-### Pump
 
-Pump: `counters.<prefix>.record.count` (number of records processed by pump)
+## NewRelic Instrumentation
 
+Tyk Gateway has been instrumented for NewRelic metrics since v2.5. Simply add the following config section to `tyk.conf` to enable the instrumentation and generation of data:
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md:40]: In addition to integrating Tyk with [New Relic Application Performance Monitoring](https://docs.newrelic.com/docs/apm/new-relic-apm/getting-started/introduction-apm/) agents, can Tyk Gateway integrate with an Open Telemetry Collector?
- https://docs.newrelic.com/docs/opentelemetry/get-started/opentelemetry-get-started-intro/
- https://github.com/open-telemetry/opentelemetry-collector-contrib

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/monitors.md [hunk 14]
```diff
@@ -1,65 +1,85 @@
...
+If you are using our [Classic Developer Portal]({{< ref "tyk-developer-portal/tyk-portal-classic/portal-events-notifications" >}}), developers registered in the portal will also receive emails about quota threshold limits being reached.
 {{< /note >}}
 
+### Setting advanced thresholds
+
+The default quota consumption monitor will be triggered at the same level of quota usage for all users. Sometimes you might want to have a more granular approach with different triggering thresholds per user or organisation. Sometimes you might want to fire the event at multiple thresholds, for example when the user hits 50%, 75% and 90% of their allowed quota.
 
-*   `monitor_user_keys`: Set to `true` to monitor individual tokens, this may result in a large amount of webhooks.
-*   `monitor_org_keys`: Set to `true` to have global organization quotas monitored.
+You can set user- or organisation-specific trigger levels by adding a `monitor` section to the user's access key ([Session Object]({{< ref "getting-started/key-concepts/what-is-a-session-object" >}})) or the organisation's session object .
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/monitors.md]: Is *user-* a typo?

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md [hunk 15]
```diff
@@ -1,55 +1,57 @@
...
+description: "Webhook event handler"
 aliases:
   - /report-monitor-trigger-events/webhooks/
 ---
 
-## Set up an Event Webhook with an API Definition
+A webhook is a mechanism for real-time, event-driven communication between different systems or applications over the internet. It is an HTTP callback, typically an HTTP POST request that occurs when something happens. Webhooks are real-time, automated and lightweight. Notifications are sent immediately when events occur without the need for the receiving service to poll.
 
-In order to add extensibility and interoperability to Tyk, a new webhook event handler has been added. This allows a fixed payload (which can be customized) to be delivered to any open endpoint.
+In the context of Tyk Gateway, webhooks are event handlers that can be registered against API Events. The webhook will be triggered when the corresponding event is fired and will send a customisable fixed payload to any open endpoint.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md]:
```suggestion
In the context of Tyk Gateway, webhooks are event handlers that can be registered against API Events. The webhook will be triggered when the corresponding event is fired and will send a customizable fixed payload to any open endpoint.
```

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md [hunk 16]
```diff
@@ -1,55 +1,57 @@
...
+
+The webhook event handler runs in its own process and so does not block the operation of the Gateway.
 
-The configuration data is pretty straight-forward, but it is important that all elements are set, as Tyk will complain if it cannot initialise the handler on startup.
+### Webhook cooldown
 
-*   `handler_meta.method`: This can be any of `GET`, `PUT`, `POST`, `PATCH` or `DELETE` and will be the method used to send the request. Request types that do not support an encoded BODY will not have the event metadata encoded as part of the request. We would advise using `POST` where possible.
+It is very likely that an `AuthFailure` event will fire on the same endpoint more than once if the requesting client is automated. If this event triggered a webhook that caused an email to be sent, then if this event occurred 10 times a second, the email recipient would be flooded with emails. In an attempt to mitigate against events such as this, you can set a cooldown timer, in the webhook handler. This prevents the webhook from being triggered again if the event is fired again within the time period specified.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:40]: Is there an example that can be added or linked to here?

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md [hunk 17]
```diff
@@ -59,64 +61,57 @@ The configuration data is pretty straight-forward, but it is important that all
...
-### Step 1: Create a Webhook
+### Local and global webhooks
 
-Select **Webhooks** from the **System Management** Menu:
+Tyk Dashboard supports the declaration of webhooks *globally* and *locally*:
+- **Global webhooks** are declared outside the API definition and linked via a *webhook id*; changes to the global webhook definition will be reflected in all APIs that reference that *webhook id*
+- **Local webhooks** are fully defined within the API definition; changes to the local webhook configuration will affect only the API within which it is defined
 
-{{< img src="/img/2.10/webhooks_menu.png" alt="Webhooks menu item" >}}
+*Global webhook definitions* are registered with the Dashboard using the [UI](#creating-a-global-webhook-definition-using-tyk-dashboard) or [Dashboard API]({{< ref "tyk-apis/tyk-dashboard-api/web-hooks" >}}) and assigned a unique *webhook id* that can be obtained via the [Dashboard API]({{< ref "" >}}) or via drop-down selection within the UI.
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md]: What link should be used for the empty link [Dashboard API]({{< ref "" >}})?

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+date: 2023-08-29T16:58:52+03:00
+tags: ["OpenTelemetry", "OpenTracing", "traces", "metrics", "logs", "observability"]
+description: Introduction to Observability in Tyk
+aliases:
+  - /product-stack/tyk-gateway/advanced-configurations/distributed-tracing/distributed-tracing-overview
+---
+
+Observability is a key concept in modern software systems that enables teams to understand the internal state of their applications and infrastructure through external outputs. It encompasses three main pillars:
+
+- Metrics: quantitative measurements of system performance and behavior over time
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md]:
```suggestion
- **Metrics**: quantitative measurements of system performance and behavior over time
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+- Logs: detailed records of events and state changes within the system
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md]:
```suggestion
- **Logs**: detailed records of events and state changes within the system
```

File: tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+- Traces: end-to-end tracking of requests as they flow through distributed systems
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/observability.md]:
```suggestion
- **Traces**: end-to-end tracking of requests as they flow through distributed systems
```

File: tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-oas.md [hunk 21]
```diff
@@ -0,0 +1,206 @@
+---
+date: 2024-07-09
+title: Webhook event handlers with Tyk OAS APIs
+tags: ["event handling", "API events", "webhook", "Tyk OAS"]
+description: "Webhook event handlers with Tyk OAS APIs"
+---
+
+[Webhooks]({{< ref "basic-config-and-security/report-monitor-trigger-events/webhooks" >}}) are event handlers that can be registered against API Events. The webhook will be triggered when the corresponding event is fired and will send a customisable fixed payload to any open endpoint.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/basic-config-and-security/report-monitor-and-trigger-events/event-webhook-tyk-oas.md]:
```suggestion
[Webhooks]({{< ref "basic-config-and-security/report-monitor-trigger-events/webhooks" >}}) are event handlers that can be registered against API Events. The webhook will be triggered when the corresponding event is fired and will send a customizable fixed payload to any open endpoint.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM @andyo-tyk , made some minor suggestions/observations

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md (hunk 6 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md]: This is a list of different event types: API event, token lifecycle event, custom event. I'll reword this to improve clarity (as I agree this is open to misinterpretation).

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md (hunk 7 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events.md]: I'll apply this change.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md (hunk 11 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/event-types.md:35]: Yes, multiple events are triggered. Theoretically this could be simplified, but would be a breaking change so needs to be carefully managed.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md (hunk 12 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md:24]: Not really in scope for this update - I've just brought this in as part of logical restructuring of existing pages. Hopefully that we have enough info here for someone who actually wants to use it - and there's a link to StatsD github earlier on the page.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md (hunk 13 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/instrumentation.md:40]: This PR is about event handling - this page was in the wrong place in the docs so I've moved it into the appropriate section (Observability). To my understanding we don't yet have OpenTelemetry metrics support.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/monitors.md (hunk 14 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/monitors.md]: This was intended to be short for "user-specific or organisation-specific". I'll rework this section slightly to improve ligibility.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md (hunk 16 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md:40]: Examples are provided in the Tyk OAS and Tyk Classic specific pages. This page is an introduction to the concepts.

File: tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md (hunk 17 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/report-monitor-trigger-events/webhooks.md]: Link added.

**Review by dcs3spp — APPROVED** [review]
